server/schedule/filter/filters.go

```diff
@@ -672,3 +673,28 @@ func (f *isolationFilter) Target(opt opt.Options, store *core.StoreInfo) bool {
 	}
 	return true
 }
+
+// cloneRegion is used to replace region.Clone in filter which call the FitRegion,
+// as FitRegion only needs partial information(startKey, endKey, peers, leader peer)
+func cloneRegion(region *core.RegionInfo, opts ...core.RegionCreateOption) *core.RegionInfo {
```

Comment:
Do not clone the `downPeers`, `pendingPeers`?  maybe rename to a suitable name?

Reply:
How about `createRegionForRuleFit(peers []*metapb.Peer, leader *metapb.Peer) *core.RegionInfo`?

Reply:
@nolouch  currently FitRegion didn't consider the downPeers and pending peers, after discussed with @disksing , I create a tracking issue https://github.com/tikv/pd/issues/2796

@disksing  good advice, updated.